uint32 counter
bool grab
float64 deltax